### PR TITLE
ci: automatic release mechanism

### PR DESCRIPTION
**Pull request by michalbali256**
Release is started locally by running the clients/vscode-hlasmplugin/start_release.sh script.
It will determine version from conventional commits, create branch release-\<version\> and push it.

Github Actions build all release-* branches and then create a prerelease. The prerelease should be tested manually and additional fixes may be pushed to the release branch, each creating a pre-release.

Once the release branch is merged to master, the CI
- automatically creates GitHub changelog and updates CHANGELOG.md in the root of project.
- updates version in package.json
- replaces "Unreleased" heading in clients/vscode-hlasmplugin/CHANGELOG.md with current version and date
- pushes a commit to master containing the version and changelog changes
- creates a new release on GitHub
- merges master into development
- pushes a commit to development, changing the version in package.json to \<version\>-NEXT and adds "Unreleased" heading to clients/vscode-hlasmplugin/CHANGELOG.md

From now on, all contributors are required to follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification. The GitHub changelog is generated automatically from all types of commits excapt `chore`. The VSIX readme is written manually.

### CODE REVIEW

File: .github/workflows/Test.yml [hunk 1]
```diff
@@ -65,11 +66,21 @@ jobs:
...
           chmod +x cmake/check_license.sh
           cmake/check_license.sh
+
+  pr-base-check:
+      name: Base branch
+      runs-on: ubuntu-18.04
+      if: github.event_name == 'pull_request'
+      steps:
+      - name: Fail, if PR base is master
+        if: contains(github.base_ref, 'master') && !contains(github.head_ref, 'development')
```
Comment (asmelko) [inline .github/workflows/Test.yml:84]: change dev to release, also i suggest to change step name so it clearly describes the step

File: clients/vscode-hlasmplugin/prepare_release.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+set -e
+set -x
+VERSION=$1
+CURRENT_BRANCH=$2
+RELEASE_NOTES=$3
+OLD_VERSION=$4
+DATE=$5
+
+{ printf "%s" "$RELEASE_NOTES"; cat ../../CHANGELOG.md; } > tmp.md
```
Comment (asmelko) [inline clients/vscode-hlasmplugin/prepare_release.sh:9]:
```suggestion
{ echo -n $RELEASE_NOTES; cat ../../CHANGELOG.md; } > tmp.md
```
?

File: clients/vscode-hlasmplugin/prepare_release.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+mv tmp.md ../../CHANGELOG.md
+
+sed -i 's/"version": ".*"/"version": "'$VERSION'"/g' package.json
+sed -i 's@\*\*\*\*Unreleased\*\*\*\*@['$VERSION'](https://github.com/michalbali256/che-che4z-lsp-for-hlasm/compare/'$OLD_VERSION'...'$VERSION') ('`date +%Y-%m-%d`')@g' CHANGELOG.md
```
Comment (asmelko) [inline clients/vscode-hlasmplugin/prepare_release.sh:13]: is there possibility to get rid of michalbali256? can you use eclipse repo?

File: clients/vscode-hlasmplugin/prepare_release.sh (hunk 2 above)
Comment (michalbali256) [reply clients/vscode-hlasmplugin/prepare_release.sh:9]: That is actually not possible, think about newlines and various special characters that may end up in release notes...

**Comment by michalbali256** [comment]
Waiting for IP check of conventional commits package
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22333

**Pull request merged**